src/openms/source/ANALYSIS/SVM/SVMWrapper.cpp

```diff
@@ -1658,6 +1656,9 @@ namespace OpenMS
       return nullptr;
     }
 
+    double temp = 0;
+    svm_problem* kernel_matrix;
```

Comment:
why does this need to be a pointer? because it otherwise allocates a lot of memory on the stack?

Reply:
svm_problem_ could also be a unique_ptr AFAIK. Copy elision will allow returning it if I understood the concept

Reply:
I don't know, just moved here. The svm_problems usually don't have to be pointers. But svm_models etc. you get back as pointers from libSVM. So, yes the correct way would be to use unique_ptrs for svm_model with custom deleter and do not use pointers for svm_problems until you pass it to libSVM I guess.

Reply:
This shoudl be addressed in a general SVM cleanup